var/spack/repos/builtin/packages/babelstream/package.py

```diff
@@ -123,108 +203,225 @@ class Babelstream(CMakePackage, CudaPackage, ROCmPackage):
             ("omp", "none", "enable Cuda backend"),
         ],
     }
-    backend_vals = ["none"]
-    for lang in backends:
-        for item in backends[lang]:
-            backend, dpdncy, descr = item
-            backend_vals.append(backend.lower())
-
-    variant("backend", values=backend_vals, default="none", description="Enable backend support")
-
-    for lang in backends:
-        for item in backends[lang]:
-            backend, dpdncy, descr = item
-            if dpdncy.lower() != "none":
-                depends_on("%s" % dpdncy.lower(), when="backend=%s" % backend.lower())
-    # this flag could be used in all required languages
-    variant("flags", values=str, default="none", description="Additional CXX flags to be provided")
 
-    # comp_impl_vals=["ONEAPI-DPCPP","DPCPP","HIPSYCL","COMPUTECPP"]
     variant(
-        "implementation",
-        values=str,
+        "ocl_backend",
+        values=("amd","cuda","intel","pocl","none"),
         default="none",
-        description="Compile using the specified SYCL compiler option",
-    )
-
-    conflicts(
-        "implementation=none",
-        when="+sycl",
-        msg="SYCL requires compiler implementation to be specified by option=",
-    )
+        when="+ocl",
+        description="Enable Backend Target for OpenCL",
+        )
+    variant(
+        "kokkos_backend",
+        values=("cuda","omp","none"),
+        default="none",
+        when="+kokkos",
+        description="Enable Backend Target for kokkos",
+        )
     conflicts(
-        "implementation=none",
-        when="+thrust",
-        msg="Which Thrust implementation to use, supported options include:\
-         - CUDA (via https://github.com/NVIDIA/thrust)\
-         - ROCM (via https://github.com/ROCmSoftwarePlatform/rocThrust)",
+        "ocl_backend=none",
+        when="+ocl",
+        msg="OpenCL implementation requires backend to be specfied by ocl_backend=",
     )
+    variant(
+        "raja_backend",
```

Comment:
Why are there separate variants for `raja_offload` and `raja_backend`?

Reply:
Thank you spotting this, I will merge them to be `raja_offload`